core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsAddress.java

```diff
@@ -77,7 +77,7 @@ public void doWriteObject(ObjectOutput output, JGroupsAddress address) throws IO
       public JGroupsAddress doReadObject(ObjectInput unmarshaller) throws IOException, ClassNotFoundException {
          try {
             org.jgroups.Address address = org.jgroups.util.Util.readAddress(unmarshaller);
-            return (JGroupsAddress) JGroupsAddressCache.fromJGroupsAddress(address);
+            return (JGroupsAddress) JGroupsAddressCache.fromJGroupsAddressUnchecked(address);
```

Comment:
Ok, I think I finally figured out what's happening. 

`JGroupsAddress.Externalizer` call was actually ok, only `JGroupsTopologyAwareAddress.Externalizer` tried to load `org.jgroups.Address` during instantiation. The reason: an implicit cast from `ExtendedUUID` to `org.jgroups.Address`. Once I changed the local variable to be a `org.jgroups.Address` as well, `LdapAuthenticationIT` stopped failing.

I updated the comments and I removed the `JGroupsAddressCache.fromJGroupsAddressUnchecked` method.